### PR TITLE
feat: goals implementation

### PR DESCRIPTION
**Pull request by alex-paca-jalafund**
## Trello Ticket

[ID-04](https://trello.com/c/RDL4oNKZ)

## What did I do?

I make all the feature of goals, update model, update request, make repository, services, interfaces, controllers, notifications, CRUD.

## Why did I do this?

In order to can manipulate the goals through API, so the user can be able to set goals and at the moment of insert incomes

## How did I do it?

Implementing all the needed features to have de CRUD of goals and the notifications reaching percentage of the goal.

## (Optional) Notes:

No notes, please if can move to the branch to make the proobs

### CODE REVIEW

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs [hunk 1]
```diff
@@ -64,15 +67,53 @@ public async Task<IActionResult> CreateIncome([FromBody] IncomeCreateRequest req
...
+                        goal.current_amount,
+                        goal.user_id
+                    );
+
+                    await _goalService.UpdateGoal(goalUpdateRequest, goal.goal_id);
+
+                    var progress = (goal.current_amount/ goal.goal_amount) * 100;
+                    if (progress >= 50 && progress < 75)
+                    {
+                        notificationMessage = "¡Has alcanzado el 50% de tu meta de ahorro!";
```
Comment (salet-gutierrez-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs]: change the message in English

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs [hunk 2]
```diff
@@ -64,15 +67,53 @@ public async Task<IActionResult> CreateIncome([FromBody] IncomeCreateRequest req
...
+                    }
+                    else if (progress >= 75 && progress < 100)
+                    {
+                        notificationMessage = "¡Has alcanzado el 75% de tu meta de ahorro!";
```
Comment (salet-gutierrez-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs]: change the message in English

File: src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs [hunk 3]
```diff
@@ -64,15 +67,53 @@ public async Task<IActionResult> CreateIncome([FromBody] IncomeCreateRequest req
...
+                    }
+                    else if (progress >= 100)
+                    {
+                        notificationMessage = "¡Has alcanzado tu meta de ahorro!";
```
Comment (salet-gutierrez-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs]: change the message in English

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

**Review by salet-gutierrez-jalafund — APPROVED** [review]
Good job

**Review by joseluis-teran-jalafund — APPROVED** [review]

**Comment by salet-gutierrez-jalafund** [comment]
[ID-04: CRUD for Goal Management](https://trello.com/c/RDL4oNKZ/39-id-04-crud-for-goal-management)

**Pull request merged**